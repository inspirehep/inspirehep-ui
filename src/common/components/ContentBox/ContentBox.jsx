import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Card } from 'antd';

import './ContentBox.scss';

class ContentBox extends Component {
  render() {
    const { title, actions, loading } = this.props;

    return (
      <div className="__ContentBox__">
        <Card title={title} loading={loading}>
          <div className="pa3">{this.props.children}</div>
          <div className="pb3">{actions}</div>
        </Card>
      </div>
    );
  }
}

ContentBox.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.node),
  children: PropTypes.node,
  title: PropTypes.string,
  loading: PropTypes.bool,
};

ContentBox.defaultProps = {
  actions: null,
  children: null,
  title: null,
  loading: false,
};

export default ContentBox;
