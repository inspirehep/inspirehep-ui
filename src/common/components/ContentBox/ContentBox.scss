.__ContentBox__ {
  .ant-card-body {
    padding: 0 !important;
  }
  .ant-card-head {
    padding: 0 1rem;
  }
  .ant-card-loading-content {
    padding: 1rem;
  }
}
