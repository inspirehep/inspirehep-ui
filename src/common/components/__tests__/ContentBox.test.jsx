import React from 'react';
import { shallow } from 'enzyme';

import ContentBox from '../ContentBox';

describe('ContentBox', () => {
  it('renders ContentBox with actions and title without loading', () => {
    const wrapper = shallow(
      <ContentBox title="Jessica Jones" actions={[<h2 key="pi">PI</h2>]}>
        <div>Defenders</div>
      </ContentBox>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ContentBox with actions and title with loading', () => {
    const wrapper = shallow(
      <ContentBox
        title="Jessica Jones"
        loading
        actions={[<h1 key="pi">PI</h1>]}
      >
        <div>Defenders</div>
      </ContentBox>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ContentBox with actions', () => {
    const wrapper = shallow(
      <ContentBox actions={[<h2 key="pi">PI</h2>]}>
        <div>Defenders</div>
      </ContentBox>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ContentBox without actions', () => {
    const wrapper = shallow(
      <ContentBox>
        <div>Defenders</div>
      </ContentBox>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ContentBox without children and without actions', () => {
    const wrapper = shallow(<ContentBox />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders ContentBox without children and with actions', () => {
    const wrapper = shallow(<ContentBox actions={[<h2 key="pi">PI</h2>]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
