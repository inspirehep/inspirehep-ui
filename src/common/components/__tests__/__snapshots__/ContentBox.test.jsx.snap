// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBox renders ContentBox with actions 1`] = `
<div
  className="__ContentBox__"
>
  <Card
    loading={false}
    title={null}
  >
    <div
      className="pa3"
    >
      <div>
        Defenders
      </div>
    </div>
    <div
      className="pb3"
    >
      <h2
        key="pi"
      >
        PI
      </h2>
    </div>
  </Card>
</div>
`;

exports[`ContentBox renders ContentBox with actions and title with loading 1`] = `
<div
  className="__ContentBox__"
>
  <Card
    loading={true}
    title="Jessica Jones"
  >
    <div
      className="pa3"
    >
      <div>
        Defenders
      </div>
    </div>
    <div
      className="pb3"
    >
      <h1
        key="pi"
      >
        PI
      </h1>
    </div>
  </Card>
</div>
`;

exports[`ContentBox renders ContentBox with actions and title without loading 1`] = `
<div
  className="__ContentBox__"
>
  <Card
    loading={false}
    title="Jessica Jones"
  >
    <div
      className="pa3"
    >
      <div>
        Defenders
      </div>
    </div>
    <div
      className="pb3"
    >
      <h2
        key="pi"
      >
        PI
      </h2>
    </div>
  </Card>
</div>
`;

exports[`ContentBox renders ContentBox without actions 1`] = `
<div
  className="__ContentBox__"
>
  <Card
    loading={false}
    title={null}
  >
    <div
      className="pa3"
    >
      <div>
        Defenders
      </div>
    </div>
    <div
      className="pb3"
    />
  </Card>
</div>
`;

exports[`ContentBox renders ContentBox without children and with actions 1`] = `
<div
  className="__ContentBox__"
>
  <Card
    loading={false}
    title={null}
  >
    <div
      className="pa3"
    />
    <div
      className="pb3"
    >
      <h2
        key="pi"
      >
        PI
      </h2>
    </div>
  </Card>
</div>
`;

exports[`ContentBox renders ContentBox without children and without actions 1`] = `
<div
  className="__ContentBox__"
>
  <Card
    loading={false}
    title={null}
  >
    <div
      className="pa3"
    />
    <div
      className="pb3"
    />
  </Card>
</div>
`;
